tools/RepoTasks/RepoTasks.Cmdlets/NewFormatPs1XmlCommand.cs

```diff
@@ -0,0 +1,135 @@
+﻿// ----------------------------------------------------------------------------------
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+namespace RepoTasks.Cmdlets
+{
+    using RemoteWorker;
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Linq;
+    using System.Management.Automation;
+
+    [Cmdlet(VerbsCommon.New, "FormatPs1Xml", SupportsShouldProcess = true)]
+    public class NewFormatPs1XmlCommand : PSCmdlet
+    {
+        [Parameter(Position = 1, ValueFromPipeline = true, Mandatory = true)]
+        public string ModulePath { get; set; }
+
+        [Parameter(Position = 2, ValueFromPipeline = true, Mandatory = false)]
+        public string[] Cmdlet { get; set; }
+
+        [Parameter(Position = 3, ValueFromPipeline = false, Mandatory = false)]
+        public string OutputPath { get; set; }
+
+        [Parameter]
+        public SwitchParameter Force { get; set; }
+
+        [Parameter]
+        public SwitchParameter OnlyMarkedProperties { get; set; }
+
+        protected override void ProcessRecord()
+        {
+            try
+            {
+                if (!Path.IsPathRooted(ModulePath))
+                {
+                    ModulePath = Path.GetFullPath(ModulePath);
```

Comment:
To resolve the path in powershell, you should use the SessionState.Path property to get a path from the given provider, see:  https://github.com/Azure/azure-powershell/blob/preview/src/ResourceManager/Websites/Commands.Websites/Strategies/GitCommand.cs#L45-L46 for an example.

Reply:
done.